packages/ez-react/src/lib/storybook-utils.tsx

```diff
@@ -1,11 +1,318 @@
+    control: { type: 'number' },
+    table: { category: 'Dimension props', defaultValue: { summary: '600px' } },
+  },
+};
+
+const axisArgTypesOptions = {
+  'xAxis.tickFormat': {
+    table: {
+      disable: true,
+    },
+  },
+  'yAxis.tickFormat': {
+    table: {
+      disable: true,
+    },
+  },
+  xAxis: {
+    table: {
+      disable: true,
+    },
+  },
+  'xAxis.domainKey': {
+    control: { type: 'object' },
+    table: { category: 'Axis Options', defaultValue: { summary: 'xValue' } },
+    description: 'Sets the X axis domain key',
+  },
+  'yAxis.domainKey': {
+    control: { type: 'object' },
+    table: { category: 'Axis Options', defaultValue: { summary: 'yValue' } },
+    description:
+      'Sets the Y axis domain key when using the default chart (! not for multi charts!)',
+    if: { arg: 'yAxis', truthy: false },
+  },
+
+  'xAxis.title': {
+    control: { type: 'text' },
+    table: { category: 'Axis Options', defaultValue: { summary: 'text' } },
+    description: 'Sets the X axis title',
+  },
+  'yAxis.title': {
+    control: { type: 'text' },
+    table: { category: 'Axis Options', defaultValue: { summary: 'text' } },
+    description:
+      'Sets the Y axis title when using the default chart (! not for multi charts!)',
+    if: { arg: 'yAxis', truthy: false },
+  },
+};
+
 export const baseChartArgTypes = {
+  ...axisArgTypesOptions,
+  ...dimensionArgTypesOptions,
+  ...animationArgTypesOptions,
+  ...paddingArgTypesOptions,
+  ...gridArgTypesOptions,
+  // we're doing this to hide the objects that we've alreay unflattened and made an input field for each of their attributes
+  scopedSlots: {
+    table: {
+      disable: true,
+    },
+  },
+  line: {
+    table: {
+      disable: true,
+    },
+  },
+  point: {
+    table: {
+      disable: true,
+    },
+  },
+  bubble: {
+    table: {
+      disable: true,
+    },
+  },
+  area: {
+    table: {
+      disable: true,
+    },
+  },
+
   isRTL: {
     control: { type: 'boolean' },
   },
+  data: {
+    control: { type: 'object' },
+    table: { defaultValue: { summary: 'Object' }, category: 'Data' },
+  },
 };
 
 export const ChartWrapper: FC<{ style?: any }> = ({ children, ...props }) => {
   return <div {...props}>{children}</div>;
 };
+/*
+ * @param args
+ * @returns { argument.att1, argument.att2, argument.att3 ...}
+ */
+export const flattenArgs = (args: Object) => {
+  let constructedArgs = {} as any;
+  for (const [key, value] of Object.entries(args)) {
+    if (
+      ['string', 'boolean', 'number'].includes(typeof value) ||
+      Array.isArray(value)
+    ) {
+      constructedArgs[key] = value;
+    } else {
+      for (const innerKey in value)
+        constructedArgs[`${key}.${innerKey}`] = value[innerKey];
+    }
+  }
+  return constructedArgs;
+};
+
+/**
+ * convert { area.stroke = "#fffff", area.strokeWidth= 0 }
+ * To {area: {stroke: '#fffff'}}
+ * @param args
+ * @returns
+ */
+export const unFlattenArgs = (args: Object) => {
```

Comment:
```suggestion
export const unFlattenArgs = (args: { [key: string]: number | string | boolean }) => {
```

Reply:
Hey mohamed args can have many different structures as an array of objects, a function or just plain objects and more.
I think it would get quite complicated.